superset-frontend/src/dashboard/util/updateComponentParentsList.js

```diff
@@ -32,9 +33,12 @@ export default function updateComponentParentsList({
 
       if (Array.isArray(currentComponent.children)) {
         currentComponent.children.forEach(childId => {
-          const child = layout[childId];
-          if (child) {
-            child.parents = parentsList; // eslint-disable-line no-param-reassign
+          if (layout[childId]) {
+            // We need to clone to avoid mutating Redux state
+            const child = cloneDeep(layout[childId]);
+            child.parents = parentsList;
+            // eslint-disable-next-line no-param-reassign
+            layout[childId] = child;
```

Comment:
This part only overwrites the `parents` so it should be enough with the spread operator.
If there's a nested field that can be violate the immutability middleware, we should fix that part instead of protecting by the deepClone.

Reply:
Makes sense. Done.